### PR TITLE
Add apiKey.js script to add API keys to database and update usage count.

### LINKED ISSUE #28: [Feature]: Add token management
### Contact Details(optional)

_No response_

### What feature are you requesting?

We want to implement token management like OpenAI, more specifically, to make the generated API key to use, like having tokens in total and drop by usage for each time, when passing the API key. It should be basic database CURD, as we've got LanceDB implemented. 

We temporarily won't need this function so it's just a proposal, and at this stage we don't want the tokens to be limited, an individual js module should work.

### PR DESCRIPTION
**Pull request by Murkeee**
- Add tools/apiKey.js
- Modified database/index.js
- Modified database/types.js

**Description**

This PR fixes #28

**Notes for Reviewers**
Hopefully this is what you want. One function for adding api key to database and another for updating the usage count. Let me know if changes need to be made.

**[Signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: tools/apiKey.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+function queryApiKeyTbl(key) {
+    const keyQuery = tbl.query()
+        .where('api_key = '+"'"+key+"'")
+        .limit(1)
+        .toArray();
+    return keyQuery
+}
+
+export async function test(key){
```
Comment (cbh778899) [inline tools/apiKey.js:30]: test functions should be removed

Comment (Murkeee) [reply tools/apiKey.js]: Sorry - Just updated that. Also I was not sure how we wanted to implement error handling.

Comment (cbh778899) [reply tools/apiKey.js]: the idea is that, if the validation is not passed or usage runs out, then the api sends corrsponding status and tells the reason, for example `res.status(401).send("User not authorized")` or `res.status(403).send("Out of usage")`

**Comment by Murkeee** [comment]
In that case should functions return status codes like this?
```javascript
if ( keyQuery.length == 0){
        await tbl.add([{api_key: key, usage: MAX_USAGE}]);
        return res.status(200).send(`Data with values: api_key: ${key}, usage: ${usage}, successfully added to table`)
    }
    else {
        return res.status(406).send("Attempted to add existing key to table");
    }
``` 
Or should that be handled differently?

**Comment by cbh778899** [comment]
> In that case should functions return status codes like this?
> 
> ```js
> if ( keyQuery.length == 0){
>         await tbl.add([{api_key: key, usage: MAX_USAGE}]);
>         return res.status(200).send(`Data with values: api_key: ${key}, usage: ${usage}, successfully added to table`)
>     }
>     else {
>         return res.status(406).send("Attempted to add existing key to table");
>     }
> ```
> 
> Or should that be handled differently?

no, i think this should be handled by functions in `/actions/*.js` files, they are specifically for req/res related actions.

**Comment by Murkeee** [comment]
👍 Got it. Thanks.

**Pull request merged**